plugins/modem-manager/fu-mm-plugin.c

```diff
@@ -32,16 +32,35 @@ fu_mm_plugin_backend_device_added(FuPlugin *plugin,
 				  GError **error)
 {
 	g_autoptr(FuDeviceLocker) locker = NULL;
+	FuDevice *dfota = NULL;
 
-	/* ignore anything from other backends, e.g. usb */
 	if (!FU_IS_MM_DEVICE(device)) {
+		dfota = fu_plugin_cache_lookup(plugin, "dfota");
```

Comment:
so dfota USB devices disappear and then reappear, but ModemManager devices don't? I think this change has the potential to break other non-DFOTA devices.

Reply:
Counterproposal; what happens if you remove the `fu_mm_device_set_inhibited(FU_MM_DEVICE(self), TRUE);` in `fu-mm-dfota.c` instead? Does MM then detect the replug?

Reply:
I'll give it a try, but iirc MM only emits `object-added` if probing is successful

Reply:
@lvoegl I'm really wondering if we want to take MM out of the equation completely here. Is there a VID/PID for the modem in runtime mode? We only appearing to use MM to do device enumeration (which it's not doing very well...) and to map the AT port to the device.

In a related request, would you be able to send me some dfota_update-capable hardware? I think it would be much better if we can work with the same hardware together.

Reply:
> @lvoegl I'm really wondering if we want to take MM out of the equation completely here. Is there a VID/PID for the modem in runtime mode? We only appearing to use MM to do device enumeration (which it's not doing very well...) and to map the AT port to the device.

I thought about that as well, but I think at least for parsing the firmware version after the update has finished we need MM, because that's more complicated than it sounds for different vendors and models.
The VID/PID stays the same before/during/after the update.

I can ask for some hardware - although if you happen to have any other Quectel device (e.g RM520) they also support it the same way.